code/modules/mob/observer/freelook/marker/signal_abilities/signal_ability.dm

```diff
@@ -147,11 +147,10 @@
 //Path to the end of the cast
 /datum/signal_ability/proc/finish_casting(var/mob/user, var/atom/target,  var/list/data)
 	//Pay the energy costs
-	if (!pay_cost(user))
+	if (!user || !pay_cost(user))
```

Comment:
if you're going to tweak this, properly abort the casting and refund the cost

Reply:
Alright